ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -259,6 +293,68 @@ protocolInfoShelley genesis protVer mbCredentials =
 
         magicTxIn = SL.TxIn magicTxInId 0
 
+    -- Register the initial staking.
+    --
+    -- This function embodies a little more logic than ideal. We might want to
+    -- move it into `cardano-ledger-specs.`
```

Comment:
Good idea. `cardano-ledger-specs` doesn't do something similar in its tests?

Let's create a ticket to move it.

Reply:
It does not - its tests generate sequences of actual transactions to set up staking